Gems/AWSCore/Code/Source/Credential/AWSDefaultCredentialHandler.cpp

```diff
@@ -64,6 +67,31 @@ namespace AWSCore
             }
         }
 
+        {
+            AZStd::lock_guard<AZStd::mutex> credentialsLock{ m_credentialMutex };
+            bool allowAWSMetadata = false;
+            AWSCoreInternalRequestBus::BroadcastResult(allowAWSMetadata, &AWSCoreInternalRequests::IsAllowedAWSMetadataQueries);
+            if (allowAWSMetadata)
+            {
+
+                if (!m_instanceProfileCredentailsProvider)
+                {
+                    SetInstanceProfileCredentialProvider(
+                        Aws::MakeShared<Aws::Auth::InstanceProfileCredentialsProvider>(AWSDEFAULTCREDENTIALHANDLER_ALLOC_TAG));
+                }
+
+                const auto ec2MetadataDisabled = Aws::Environment::GetEnv(AWS_EC2_METADATA_DISABLED);
+                if (Aws::Utils::StringUtils::ToLower(ec2MetadataDisabled.c_str()) != "true")
```

Comment:
Why have the double check, surely this fails internally and you get null credentials? We also have a double call to SetInstanceProfileCredentialProvider so not 100% clear of flow/

Is one setting up for default? where as the other allows you to dynamically add?

Reply:
> Why have the double check, surely this fails internally and you get null credentials?

If you mean "won't the `InstanceProfileCredentialsProvider` deal with checking this env var?", I thought so too, but the AWS SDK is checking for this env var at the [AWSCredentialsProviderChain](https://github.com/aws/aws-sdk-cpp/blob/main/aws-cpp-sdk-core/source/auth/AWSCredentialsProviderChain.cpp#L74) level, which this `AWSDefaultCredentialHander` is acting in lieu of. The [InstanceProfileCredentialsProvider](https://github.com/aws/aws-sdk-cpp/blob/main/aws-cpp-sdk-core/source/auth/AWSCredentialsProvider.cpp#L220-L274) itself doesn't include any direct checks for this env var, so while calls to IMDS will still fail, it seems it'll still be attempted unless we do this check and don't return it at all.

> We also have a double call to SetInstanceProfileCredentialProvider so not 100% clear of flow

I'm less clear on this one myself, but [our logic for `ProfileConfigFileAWSCredentialsProvider`](https://github.com/o3de/o3de/blob/development/Gems/AWSCore/Code/Source/Credential/AWSDefaultCredentialHandler.cpp#L51-L58) suggests that the configuration settings may change between Initializing of the `AWSDefaultCredentialHander` and calls to `AWSDefaultCredentialHandler::GetCredentialHandlerOrder()` (why else would there be a check comparing the current & previous profile names?). So the check on line 77 is to cover the case where `AllowAWSMetadataQueries` was changed to "true" on reload so that we don't return a null `m_instanceProfileCredentailsProvider`. However, if this isn't something to worry about or I've grokked the profile provider logic wrong please LMK, will gladly remove unnecessary checks.

Reply:
I suppose we could remove the check & set in `InitCredentialsProviders()` instead of duplicating it, though it's not clear to me why it's duplicated for `ProfileConfigFileAWSCredentialsProvider` (maybe no reason?)